### DISCUSSION TITLE
Env variables can't use in material git branch.

### DISCUSSION THREAD

**Discussion by zszen** [gocd/gocd#12594] [Q&A] [answered]
![image](https://github.com/gocd/gocd/assets/1890538/7478a51b-e6ec-430a-8ad0-14af020610b4)

![image](https://github.com/gocd/gocd/assets/1890538/d5451d0c-176c-45d9-b532-ed36d6218efe)

![image](https://github.com/gocd/gocd/assets/1890538/784ee980-8369-4860-a87d-2a990a3e0798)

**Comment by chadlwilson** [comment, marked as answer]
No, and this is by design. Materials in GoCD are independent of pipelines, and environment variables are a process-runner/OS-level construct that are only known at run time. Not all materials in GoCD assume use of a separate OS-level process call, so dynamic environment variables don't make sense there either.

In practice you can use environment variables only in tasks; when they are run on agents.

You *might* be able to use [pipeline parameters](https://docs.gocd.org/current/configuration/admin_use_parameters_in_configuration.html) instead if you are using pipeline templates (with multiple pipelines for different branches), but whether that makes sense depends on what you are trying to achieve.

**Reply by zszen** [reply]
Thank you for reply. I will goto learn about pipeline templates.
